homeassistant/components/lamarzocco/update.py

```diff
@@ -0,0 +1,115 @@
+from dataclasses import dataclass
+from typing import Any
+
+from lmcloud import LMCloud as LaMarzoccoClient
+from lmcloud.const import LaMarzoccoUpdateableComponent
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityDescription,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import LaMarzoccoUpdateCoordinator
+from .entity import LaMarzoccoEntity, LaMarzoccoEntityDescription
+
+
+@dataclass(frozen=True, kw_only=True)
+class LaMarzoccoUpdateEntityDescription(
+    LaMarzoccoEntityDescription,
+    UpdateEntityDescription,
+):
+    """Description of an La Marzocco updates."""
+
+    current_fw_fn: Callable[[LaMarzoccoClient], str]
+    latest_fw_fn: Callable[[LaMarzoccoClient], str]
+    component: LaMarzoccoUpdateableComponent
+
+
+ENTITIES: tuple[LaMarzoccoUpdateEntityDescription, ...] = (
+    LaMarzoccoUpdateEntityDescription(
+        key="machine_firmware",
+        translation_key="machine_firmware",
+        device_class=UpdateDeviceClass.FIRMWARE,
+        icon="mdi:cloud-download",
+        current_fw_fn=lambda lm: lm.firmware_version,
+        latest_fw_fn=lambda lm: lm.latest_firmware_version,
+        component=LaMarzoccoUpdateableComponent.MACHINE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    LaMarzoccoUpdateEntityDescription(
+        key="gateway_firmware",
+        translation_key="gateway_firmware",
+        device_class=UpdateDeviceClass.FIRMWARE,
+        icon="mdi:cloud-download",
+        current_fw_fn=lambda lm: lm.gateway_version,
+        latest_fw_fn=lambda lm: lm.latest_gateway_version,
+        component=LaMarzoccoUpdateableComponent.GATEWAY,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Create update entities."""
+
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        LaMarzoccoUpdateEntity(coordinator, description)
+        for description in ENTITIES
+        if description.supported_fn(coordinator)
+    )
+
+
+class LaMarzoccoUpdateEntity(LaMarzoccoEntity, UpdateEntity):
+    """Entity representing the update state."""
+
+    entity_description: LaMarzoccoUpdateEntityDescription
+    _attr_supported_features = UpdateEntityFeature.INSTALL
+
+    def __init__(
+        self,
+        coordinator: LaMarzoccoUpdateCoordinator,
+        description: LaMarzoccoUpdateEntityDescription,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator, description)
+        self._attr_in_progress = False
```

Comment:
Is default, so you can remove the whole constructor

Reply:
already on it 😉